### DISCUSSION TITLE
Koishi v4 精品问答集

### DISCUSSION THREAD

**Discussion by idlist** [koishijs/koishi#500] [General]
虽然没有什么要写的，但是好像这个会是第 500 个 issue，占个号（不是）

**Comment by ilharp** [comment]
前任： [Koishi v3 精品问答集（#270）](https://github.com/koishijs/koishi/discussions/270)

**Comment by idlist** [comment]
Q: 在 v3 的时候复写指令描述的写法在 v4 忽然不管用了，怎么办！

A: 这是因为一些功能改热了而导致的问题。一个解决方法是：

```js
ctx.on('command-added', command => {
  if (command.name == 'schedule') ctx.command('schedule', '俺的定时')
})
```
如果想要复写 `usage`，那么需要额外 `await sleep(0)`：
```js
ctx.on('command-added', async (command) => {
  await sleep(0); // const { sleep } = require('koishi')
  if (command.name == 'teach') ctx.command('teach', '教学').usage('太长了，砍掉！')
})
```
在非热重载环境下，复写指令配置项（`command` 3 参数）和将指令挂在其他指令之下的操作使用旧语法仍然生效。

听说 4.4 的模板服务也能解决这个问题，但现在这还是个饼，先用着这个吧！

**Reply by idlist** [reply]
备注：写这个问题的时候是v4.3，现在v4.5下这一招又不管用了！但是有新的管用的方法，待我咕咕咕咕咕咕咕

**Comment by ilharp** [comment]
**本问题现已置于文档中**

<https://koishi.js.org/manual/advanced/server.html>

---

Q: 我的控制台看起来加载完成了！可以切换暗色模式了！但没有任何页面显示出来！

A: 最有可能的一种情况是，你的 Http 连接正常，但位于 `/status` 路由的 WebSocket 无法连接。

如果你使用了反向代理服务，你需要手动配置这个路由。

以 Nginx 为例，在 `server` 节点中加入：

```conf
location /status {
    proxy_pass http://127.0.0.1:8080/status;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
}
```

*在配置路由前，记得检查你的 WS 路由是否为默认的 `/status`。

**Reply by purerosefallen** [reply]
Caddy 就没这种破事儿。

**Reply by ilharp** [reply]
呜呜呜，羡慕 Caddy

**Reply by shigma** [reply]
我以为这是 nginx 的常识。不比记得 Caddy 这五个字母的拼写更复杂的破事。

**Reply by ilharp** [reply]
> 我以为这是 nginx 的常识。不比记得 Caddy 这五个字母的拼写更复杂的破事。

呜呜呜，呜呜呜呜呜

**Reply by simon300000** [reply]
也可以让ws fallback到一个更加传统的pulling

**Reply by shigma** [reply]
> 呜呜呜，呜呜呜呜呜

我的意思是，nginx 人知道这个就跟 Caddy 人知道 Caddy 的拼写一样，根本没资格用这点嘲讽别人选的框架。
